extensions/librdkafka/PublishKafka.cpp

```diff
@@ -541,6 +543,11 @@ void PublishKafka::onSchedule(const std::shared_ptr<core::ProcessContext> &conte
   conn_ = utils::make_unique<KafkaConnection>(key_);
   configureNewConnection(context);
 
+  std::string message_key_field;
+  if (context->getProperty(MessageKeyField.getName(), message_key_field)) {
+    logger_->log_error("The %s property is set. This property is DEPRECATED and has no effect; please use Kafka Key instead.", MessageKeyField.getName());
```

Comment:
I would prefer this to be a warning, not an error

Reply:
I would go with error, as it uses a property that never worked and is only preserved to allow the processor to operate instead of fizzling out in onSchedule, personally I would even go for removing it, but I do not know how prevalent its usage is

Reply:
Having the property set is likely a mistake, but it doesn't cause data loss or failures down the line. Seeing this message doesn't mean a change in behavior from the happy path, only that the flow is incorrect, but still runnable.

Reply:
I'm fine with logging it on either warning or error level, so we'll need a 4th vote to decide. :)

My reasoning for an error log was that if the user has set the MessageKeyField property, then their flow is not working as intended.  E.g. they may have wanted to set the message key to the value of a particular flow file attribute, but it is set to the flow file UUID instead -- this can cause difficult-to-debug problems down the line.

Reply:
I tend to agree with the error approach. Especially as this is only going to be logged once, so won't spam the log with error messages.

Reply:
ok